### PR TITLE
Fixes and additions to LFSC signature

### PR DESCRIPTION
**Pull request by ajreynol**
Includes:
- Proper printing of various FP terms and constants.
- Distinguishing "variants" of overloaded symbols, in case a user declares 2 symbols with the same name.
- Miscellaneous fixes for printing terms.

### CODE REVIEW

**Comment by yoni206** [comment]
Thanks for all of this.
Any chance you could split it to several PRs? This would make it easier for me to properly review. A suggestion:

PR 1: variants of overloading + Miscellaneous fixes in printing.
PR 2: properly printing FP and Datatypes
PR 3: sequences signature.

Could that work?

**Comment by ajreynol** [comment]
I've done https://github.com/cvc5/cvc5/pull/8220, which is PR 3.

**Comment by ajreynol** [comment]
https://github.com/cvc5/cvc5/pull/8222 fixes datatype printing, then the remainder can be reviewed in this PR after that one + rebasing.

File: src/proof/lfsc/lfsc_node_converter.cpp [hunk 1]
```diff
@@ -86,6 +89,7 @@ Node LfscNodeConverter::postConvert(Node n)
 {
   NodeManager* nm = NodeManager::currentNM();
   Kind k = n.getKind();
+  Assert(k != MATCH);
```
Comment (yoni206) [inline src/proof/lfsc/lfsc_node_converter.cpp:92]: Please document why is that.

**Review by yoni206 — APPROVED** [review]
One minor comment.

**Pull request merged**